### PR TITLE
Show armor

### PR DESCRIPTION
**Pull request by ataulien**
Changes:
 - Fixes Skeletal-Meshes broken when the source files supply their own skeleton
 - Disables physics on all characters not controlled by the player for performance reasons,
 - Dresses the Characters with the Armor equipped inside their instance constructor

### CODE REVIEW

File: src/scripting/daedalus/DaedalusVMForGameWorld.cpp [hunk 1]
```diff
@@ -292,6 +292,24 @@ namespace REGoth
       bs::INT32 bodyTexIndex = popIntValue();
       bs::String bodyMesh    = popStringValue();
 
+      // If an armor is set here, we need to replace the body mesh from the input parameters with the
+      // visual settings from inside the armor instance
+      if (armorInstance != -1)
+      {
+        // TODO: Original Gothic adds the Armor straight to the inventory. We just create a
+        // temporary instance to extract the visual information from it
+        ScriptObjectHandle armorSObj = getInstance(armorInstance);
```
Comment (mworchel) [inline src/scripting/daedalus/DaedalusVMForGameWorld.cpp]: `armorInstance` is a `bs::UINT32` but `getInstance` expects a `bs::String`, so it doesn't compile for me due to this line

File: src/main_WorldViewer.cpp [hunk 2]
```diff
@@ -32,11 +32,11 @@ class REGothWorldViewer : public REGoth::REGothEngine
     {
       world = GameWorld::importZEN(WORLD);
 
-      HCharacter hero = world->insertCharacter("PC_HERO", "STARTPOINT");
+      HCharacter hero = world->insertCharacter("PC_HERO", "START_DRAGONISLAND");
```
Comment (mworchel) [inline src/main_WorldViewer.cpp]: `"START_DRAGONISLAND"` only works for Dragon Island I suppose? :D

File: src/components/CharacterAI.cpp [hunk 3]
```diff
@@ -76,6 +86,7 @@ namespace REGoth
 
     mTurnDirection = TurnDirection::Left;
 
+    // return mVisual->tryPlayTransitionAnimationTo("T_RUNTURNL");
```
Comment (mworchel) [inline src/components/CharacterAI.cpp]: Did you try to run the turning animations but it doesn't work? Is this code supposed to stay commented out or even stay at all?

File: src/components/CharacterAI.cpp [hunk 4]
```diff
@@ -85,6 +96,7 @@ namespace REGoth
 
     mTurnDirection = TurnDirection::Right;
 
+    // return mVisual->tryPlayTransitionAnimationTo("T_RUNTURNR");
```
Comment (mworchel) [inline src/components/CharacterAI.cpp]: Same as line 89

File: src/components/CharacterAI.cpp [hunk 5]
```diff
@@ -97,6 +109,11 @@ namespace REGoth
 
   bool CharacterAI::stopTurning()
   {
+    // if (mTurnDirection != TurnDirection::None)
+    // {
+    //   mVisual->playAnimation(mVisual->findAnimationClip("S_RUN"));
```
Comment (mworchel) [inline src/components/CharacterAI.cpp]: Same as line 89

File: src/components/CharacterAI.cpp (hunk 3 above)
Comment (ataulien) [reply src/components/CharacterAI.cpp]: The code is correct, but there is a bug with the animation importer which makes turning animations look weird. I can remove those lines from the PR. 
I agree that commented out code without any note as to *why* it was commented out isn't great.

File: src/scripting/daedalus/DaedalusVMForGameWorld.cpp (hunk 1 above)
Comment (ataulien) [reply src/scripting/daedalus/DaedalusVMForGameWorld.cpp]: Weird, did I forget to commit something?

File: src/main_WorldViewer.cpp (hunk 2 above)
Comment (ataulien) [reply src/main_WorldViewer.cpp]: A method to get the world starting point from the `GameWorld` would be nice to have I guess.

File: src/scripting/daedalus/DaedalusVMForGameWorld.cpp (hunk 1 above)
Comment (ataulien) [reply src/scripting/daedalus/DaedalusVMForGameWorld.cpp]: There seems to be more missing code, actually. Now, I wonder where that went...

Comment (ataulien) [reply src/scripting/daedalus/DaedalusVMForGameWorld.cpp]: Can't find it in stashes, it's not on other branches, guess I'lll write that again.

Comment (ataulien) [reply src/scripting/daedalus/DaedalusVMForGameWorld.cpp]: Actually I found those commits! Seems like I forgot to push them before merging https://github.com/REGoth-project/REGoth-bs/pull/21...

File: src/scripting/daedalus/DaedalusVMForGameWorld.hpp [hunk 6]
```diff
@@ -93,7 +99,7 @@ namespace REGoth
     public:
       REGOTH_DECLARE_RTTI(DaedalusVMForGameWorld);
 
-    public:                              // FIXME: RTTI protected
+    public:                                // FIXME: RTTI protected
```
Comment (ataulien) [inline src/scripting/daedalus/DaedalusVMForGameWorld.hpp]: We should take care of the remaining RTTI-protected FIXME in another PR. It's not straightforward for bs::IReflectable. We also befriend bs::SceneObject inside REGOTH_DECLARE_RTTI, which is only correct for components.

Comment (mworchel) [reply src/scripting/daedalus/DaedalusVMForGameWorld.hpp]: We probably need separate `DECLARE` and `DEFINE` macros for `IReflectable` and  `Component` (and probably also a seperate one for pure abstract `IReflectables`). Doing it for `IReflectable` is pretty straight-forward (see `RaycastShadowSampler` in the shadow-sampling branch) and could also be encapsulated in a macro.

File: src/main_WorldViewer.cpp (hunk 2 above)
Comment (mworchel) [reply src/main_WorldViewer.cpp]: Is this something one could easily get from the waynet? Like does it always contain the string `"START"` or something?

**Comment by mworchel** [comment]
Compiles and works 👍 
What about the start point stuff?

**Comment by ataulien** [comment]
That's something for an other PR. Right now the player is just dropped at position (0,0,0) if the startpoint can't be found. I guess that would be another good-first-issue :smile:

**Pull request merged**